pandas/core/arrays/categorical.py

```diff
@@ -639,11 +639,13 @@ def _from_inferred_categories(cls, inferred_categories, inferred_codes,
         return cls(codes, dtype=dtype, fastpath=True)
 
     @classmethod
-    def from_codes(cls, codes, categories, ordered=False):
+    @deprecate_kwarg(old_arg_name='categories', new_arg_name=None)
+    @deprecate_kwarg(old_arg_name='ordered', new_arg_name=None)
```

Comment:
not sure this gives u the replacement message
rather than this just check in the function